extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -213,9 +213,9 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 			const sourceBook = this.books.find(book => book.bookPath === movingElement.book.root);
 			const targetBook = this.books.find(book => book.bookPath === updateBook.book.root);
 			this.bookTocManager = new BookTocManager(targetBook, sourceBook);
-			// remove watch on toc file from both books.
+			// remove watch on toc file from source book.
 			if (sourceBook) {
-				fs.unwatchFile(movingElement.tableOfContentsPath);
+				fs.unwatchFile(sourceBook.tableOfContentsPath);
```

Comment:
Having it call setFileWatcher with just sourceBook but then needing to know that it's actually watching tableOfContentsPath for the unwatch isn't really intuitive. What if someone changes setFileWatcher to watch a different file? There's no way to know that this is just going to stop doing anything unless the person happens to know/catch it.

What would be really cool is if you had setFileWatcher return a disposable - that when disposed would automatically unhook the file watcher. That would be a lot safer since all the logic for hooking/unhooking would be done in the same place. 

See https://github.com/microsoft/azuredatastudio/blob/main/extensions/azdata/src/azdata.ts#L38 for a simple example of doing this - the class implements vscode.Disposable and then takes care of cleaning up the session when it's disposed.

Reply:
I'll implement this next week, since I'm going to be adding more operations to the bookTocManager